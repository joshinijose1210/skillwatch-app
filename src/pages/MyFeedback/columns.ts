import {
    DateFormatter,
    FeedbackStatusFormatter,
    FeedbackTagFormatter,
    FullNameReceivedFeedbackFormatter,
    FullNameSubmittedFeedbackFormatter,
    ShowFeedbackDataFormatter,
    ViewFeedbackButton
} from '@components';
import { ViewReceivedFeedbackButton } from '@components/columnFormatters/ViewReceivedFeedbackButton';
import { TableColumnConfig } from '@medly-components/core';

export const SubmittedFeedbackCols: TableColumnConfig[] = [
    { title: 'Date', field: 'date', sortable: true, component: DateFormatter, fraction: 0.05 },
    { title: 'Feedback About', field: 'empFirstName', component: FullNameSubmittedFeedbackFormatter, fraction: 0.225 },
    { title: 'Feedback', field: 'feedbackData', component: ShowFeedbackDataFormatter, fraction: 0.475 },
    { title: 'Feedback Type', field: 'feedbackType', component: FeedbackTagFormatter, fraction: 0.125 },
    { title: 'Status', field: 'isDraft', component: FeedbackStatusFormatter, fraction: 0.125 },
    { title: 'Action', field: 'action', component: ViewFeedbackButton, fraction: 0.125 }
];

export const ReceivedFeedbackCols: TableColumnConfig[] = [
    { title: 'Date', field: 'date', sortable: true, component: DateFormatter, fraction: 0.05 },
    { title: 'Feedback From', field: 'empFirstName', component: FullNameReceivedFeedbackFormatter, fraction: 0.225 },
    { title: 'Feedback', field: 'feedbackData', component: ShowFeedbackDataFormatter, fraction: 0.475 },
    { title: 'Feedback Type', field: 'feedbackType', component: FeedbackTagFormatter, fraction: 0.2 },
    { title: 'Action', field: 'action', component: ViewReceivedFeedbackButton, fraction: 0.2 }
];
