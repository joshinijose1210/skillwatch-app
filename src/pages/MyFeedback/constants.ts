export const FeedbackPageTabValues = ['index', 'submitted', 'received', 'inbox', 'sent'];
