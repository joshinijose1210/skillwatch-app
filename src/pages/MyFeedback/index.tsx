import MyFeedback from './MyFeedback';

export default MyFeedback;
