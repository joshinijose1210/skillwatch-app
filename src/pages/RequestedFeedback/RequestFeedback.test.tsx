import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { routeConstants } from '@constants';
import MyFeedback from '@pages/MyFeedback';
import { RTKRender, cleanup, fireEvent, screen, waitFor } from '@utils/test-utils';
import { requestFeedbackMockData } from '@mocks/requestFeedback';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });
const mockNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
    useLocation: jest.fn()
}));

beforeAll(() => {
    const { url, data } = requestFeedbackMockData.feedback;
    mock.onGet(url).replyOnce(200, data);
    const { url: employeeUrl, data: employeeData } = requestFeedbackMockData.employee;
    mock.onGet(employeeUrl).replyOnce(200, employeeData);
});

beforeEach(mock.reset);

afterEach(cleanup);

describe('RequestedFeedback tabs', () => {
    it('should take snapshot and render properly', async () => {
        const { container } = RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('inbox')); //Request Received tab should be open before snapshot is taken
        expect(container).toMatchSnapshot();
        const { url, data } = requestFeedbackMockData.feedback;
        mock.onGet(url).replyOnce(200, data);
        const { url: employeeUrl, data: employeeData } = requestFeedbackMockData.employee;
        mock.onGet(employeeUrl).replyOnce(200, employeeData);
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        expect(screen.getAllByText('Request Feedback')[0]).toBeInTheDocument();
    });

    it('should tab change propely', async () => {
        RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('sent'));
        expect(await screen.findAllByText('Feedback From')).toHaveLength(2);
        fireEvent.click(screen.getByTestId('inbox'));
        expect(await screen.findAllByText('Requested From')).toHaveLength(2);
    });

    it('should redirect to add feedback page when clicked on add-feedback', () => {
        const { url, data } = requestFeedbackMockData.feedback;
        mock.onGet(url).replyOnce(200, data);
        RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('inbox'));
        fireEvent.click(screen.getAllByText('Request Feedback')[0]);
        expect(mockNavigate).toHaveBeenCalledWith(
            `${routeConstants.threeSixtyDegreeFeedback}${routeConstants.requestFeedback}/undefined/Request`,
            {
                state: { requestId: undefined }
            }
        );
    });

    it('should redirect when click on view', async () => {
        const { url, data } = requestFeedbackMockData.feedback;
        mock.onGet(url).replyOnce(200, data);
        const { url: employeeUrl, data: employeeData } = requestFeedbackMockData.employee;
        mock.onGet(employeeUrl).replyOnce(200, employeeData);
        RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('inbox'));
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getAllByText('View')[0]);
        expect(mockNavigate).toHaveBeenCalledWith(
            `${routeConstants.threeSixtyDegreeFeedback}${routeConstants.requestFeedback}/undefined/View-Requested-Feedback`,
            {
                state: {
                    ...requestFeedbackMockData.navigationState.state,
                    action: 'View'
                }
            }
        );
    });

    it('should filter work properly', async () => {
        const { url, url2, url3, url4, data } = requestFeedbackMockData.feedback;
        mock.onGet(url)
            .replyOnce(200, data)
            .onGet(url2)
            .replyOnce(200, data)
            .onGet(url3)
            .replyOnce(200, {})
            .onGet(url4)
            .replyOnce(200, data);
        const { url: employeeUrl, data: employeeData } = requestFeedbackMockData.employee;
        mock.onGet(employeeUrl).replyOnce(200, employeeData);
        RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('inbox'));
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getByTestId('employeeDropdown'));
        fireEvent.click(screen.getAllByText('M A (SR0067)')[0]);
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        fireEvent.click(screen.getByTestId('feedbackToDropdown'));
        fireEvent.click(screen.getByText('U H (SR0069)'));
        mock.onGet(employeeUrl).replyOnce(200, employeeData);
        expect(await screen.findByText('No result')).toBeInTheDocument();
        fireEvent.change(screen.getByTestId('statusDropdown'), { target: { value: 'Complete' } });
        fireEvent.click(screen.getByText('Completed'));
        await waitFor(() => expect(screen.queryAllByText('View')[0].innerHTML).toEqual('View'), { timeout: 3000 });
    });

    it('should sorting work propely', async () => {
        const { url, sortUrl, data } = requestFeedbackMockData.feedback;
        mock.onGet(url).replyOnce(200, data).onGet(sortUrl).replyOnce(200, data);
        RTKRender(<MyFeedback />);
        fireEvent.click(screen.getByTestId('inbox'));
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        const spanElement = screen.getByText('Date');
        fireEvent.click(spanElement);
        await waitFor(() => screen.findAllByText('View'), { timeout: 3000 });
        await waitFor(() => expect(screen.queryAllByText('View')[0].innerHTML).toEqual('View'), { timeout: 3000 });
    });
});
