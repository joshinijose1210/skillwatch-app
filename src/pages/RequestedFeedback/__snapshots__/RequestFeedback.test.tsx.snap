// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestedFeedback tabs should take snapshot and render properly 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          360-Degree Feedback
        </h3>
      </div>
      <div
        class="sc-w8e7z6-3 rCDhx"
      >
        <button
          class="sc-1k2druj-0 dsEKPA sc-w8e7z6-1"
          edges="square"
          type="button"
          variant="outlined"
        >
          <span
            class="sc-1ntgl3l-0 eOsHHD"
            textalign="initial"
            textcolor="inherit"
            textvariant="button1"
          >
            Request Feedback
          </span>
        </button>
        <button
          class="sc-1k2druj-0 eLgfbc sc-w8e7z6-1"
          edges="square"
          type="button"
          variant="solid"
        >
          <span
            class="sc-1ntgl3l-0 eOsHHD"
            textalign="initial"
            textcolor="inherit"
            textvariant="button1"
          >
            Add Feedback
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-1h7claq-0 jdWERm"
    >
      <div
        class="sc-1fhg5t3-2"
      >
        <div
          class="sc-11tbpdo-0 fpmwxR sc-tkiih6-14 sc-1fhg5t3-4 gSveWN ebGFtX"
          data-testid="employeeDropdown"
          id="requestedfrom-wrapper"
          minwidth="25rem"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 UfceN"
            id="requestedfrom-input-wrapper"
            minwidth="25rem"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="requestedfrom-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  data-tooltip-id="requestedfrom-input"
                  errortext=""
                  id="requestedfrom-input"
                  inputsize="M"
                  placeholder="Select Employee Name"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="requestedfrom-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Requested From
                </label>
              </div>
              <div
                class="sc-4vbg19-0 hFRmRP"
                id="requestedfrom-count"
              >
                <svg
                  class="sc-1pf3x8a-0 klDMsE"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
            </div>
            
          </div>
        </div>
        
        <div
          class="sc-11tbpdo-0 fpmwxR sc-tkiih6-14 gSveWN"
          data-testid="feedbackToDropdown"
          id="feedbackabout-wrapper"
          minwidth="25rem"
          variant="outlined"
        >
          <div
            class="sc-17d4s39-0 UfceN"
            id="feedbackabout-input-wrapper"
            minwidth="25rem"
          >
            <div
              class="sc-19px6zp-0 BrSGZ"
              filled="[object Object]"
              fusion="[object Object]"
              minrows="1"
              outlined="[object Object]"
              textvariant="[object Object]"
              variant="outlined"
            >
              <div
                class="sc-lhr232-0 btNOKw"
                variant="outlined"
              >
                <input
                  aria-describedby="feedbackabout-helper-text"
                  autocomplete="off"
                  class="sc-hb99wz-0 jaEbhp"
                  errortext=""
                  id="feedbackabout-input"
                  inputsize="M"
                  placeholder="Select Employee Name"
                  type="text"
                  value=""
                  variant="outlined"
                />
                <label
                  class="sc-1du51u0-0 eMZrST"
                  for="feedbackabout-input"
                  inputwidth="0"
                  variant="outlined"
                >
                  Feedback About
                </label>
              </div>
              <div
                class="sc-4vbg19-0 hFRmRP"
                id="feedbackabout-count"
              >
                <svg
                  class="sc-1pf3x8a-0 klDMsE"
                  fill="none"
                  height="1em"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                    fill="#000"
                  />
                </svg>
              </div>
            </div>
            
          </div>
        </div>
        
      </div>
      <div
        class="sc-1hcsztp-0 fKSUGy sc-tkiih6-13 cecHDR"
        id="status-wrapper"
        minwidth="25rem"
        variant="outlined"
      >
        <div
          class="sc-17d4s39-0 UfceN"
          id="status-input-wrapper"
          minwidth="25rem"
        >
          <div
            class="sc-19px6zp-0 BrSGZ"
            filled="[object Object]"
            fusion="[object Object]"
            minrows="1"
            outlined="[object Object]"
            textvariant="[object Object]"
            variant="outlined"
          >
            <div
              class="sc-lhr232-0 btNOKw"
              variant="outlined"
            >
              <input
                aria-describedby="status-helper-text"
                autocomplete="off"
                class="sc-hb99wz-0 jaEbhp"
                data-testid="statusDropdown"
                errortext=""
                id="status-input"
                inputmode="none"
                inputsize="M"
                placeholder="Select"
                type="text"
                value="All"
                variant="outlined"
              />
              <label
                class="sc-1du51u0-0 eMZrST"
                for="status-input"
                inputwidth="0"
                variant="outlined"
              >
                Status
              </label>
            </div>
            <svg
              class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                fill="#000"
              />
            </svg>
          </div>
          
        </div>
      </div>
    </div>
    <div
      aria-label="Closed style tabs"
      class="sc-13bppbd-0 sc-tkiih6-28 sc-1fhg5t3-0 dErHOx hlXJvb"
      id="medly-tabs"
    >
      <div
        class="sc-12mpb33-0 dYQjJT"
        id="medly-tabs-list"
        tabsize="M"
        variant="outlined"
      >
        <button
          aria-controls="panel-submitted"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 fpIKmc"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="submitted"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="M"
          totaltabs="4"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.296 5.704c0 .56.45 1 1 1h5.59l-10.89 10.89a.996.996 0 1 0 1.41 1.41l10.89-10.89v5.59c0 .55.45 1 1 1s1-.45 1-1v-8c0-.55-.45-1-1-1h-8c-.55 0-1 .45-1 1z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="submitted-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Feedback Given
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-received"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 fpIKmc"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="received"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="M"
          totaltabs="4"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.008 5.001a.996.996 0 0 0-1.41 0l-10.89 10.88v-5.59c0-.55-.45-1-1-1s-1 .45-1 1v8c0 .55.45 1 1 1h8c.55 0 1-.45 1-1s-.45-1-1-1h-5.59l10.89-10.89c.38-.38.38-1.02 0-1.4z"
              fill="#000"
              fill-opacity=".54"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="received-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Feedback Received
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-sent"
          aria-selected="false"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 fpIKmc"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          data-testid="sent"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="sent"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="-1"
          tabsize="M"
          totaltabs="4"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M4.5 9h2c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1zm2 5h-2c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1h2c.55 0 1 .45 1 1v2c0 .55-.45 1-1 1zm0 5h-2c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1h2c.55 0 1 .45 1 1v2c0 .55-.45 1-1 1zm13-5h-10c-.55 0-1-.45-1-1v-2c0-.55.45-1 1-1h10c.55 0 1 .45 1 1v2c0 .55-.45 1-1 1zm-10 5h10c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1h-10c-.55 0-1 .45-1 1v2c0 .55.45 1 1 1zm-1-11V6c0-.55.45-1 1-1h10c.55 0 1 .45 1 1v2c0 .55-.45 1-1 1h-10c-.55 0-1-.45-1-1z"
              fill="#000"
              fill-opacity=".54"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="sent-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Request Sent
              </span>
            </div>
          </div>
        </button>
        <button
          aria-controls="panel-inbox"
          aria-selected="true"
          bgcolor="[object Object]"
          bordercolor="[object Object]"
          class="sc-1kcg570-4 ewhzXK"
          countbgcolor="[object Object]"
          countborderradius="2.5rem"
          countcolor="#ffffff"
          data-testid="inbox"
          helpertextcolor="[object Object]"
          iconbgcolor="[object Object]"
          iconcolor="[object Object]"
          id="inbox"
          label="[object Object]"
          labelcolor="[object Object]"
          role="tab"
          solid="[object Object]"
          tabbackground="WHITE"
          tabindex="0"
          tabsize="M"
          totaltabs="4"
          type="button"
          variant="outlined"
        >
          <svg
            class="sc-1pf3x8a-0 kdcvez"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M19 16.75l1.29 1.29c.63.63.19 1.71-.71 1.71H4.41c-.89 0-1.33-1.08-.7-1.71L5 16.75v-6C5 7.4 7.36 4.6 10.5 3.92V2.75c0-.83.67-1.5 1.5-1.5s1.5.67 1.5 1.5v1.17c3.14.68 5.5 3.48 5.5 6.83v6zm-5.01 4.01c0 1.1-.89 1.99-1.99 1.99s-1.99-.89-1.99-1.99h3.98zM11 15.75v-2h2v2h-2zm1-4c.55 0 1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1v2c0 .55.45 1 1 1z"
              fill="#000"
              fill-opacity=".54"
              fill-rule="evenodd"
            />
          </svg>
          <div
            class="sc-1kcg570-5 kFiIrt"
          >
            <div
              class="sc-1kcg570-6 jobpmu"
            >
              <span
                class="sc-1ntgl3l-0 hGagSU sc-1kcg570-1 jILKEJ"
                id="inbox-label"
                tabsize="M"
                textalign="initial"
                textcolor="inherit"
              >
                Request Received
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        aria-labelledby="inbox"
        class="sc-x0gfjw-0 bLZBSc"
        id="medly-tabs-panel-inbox"
        role="tabpanel"
        tabindex="0"
      />
    </div>
    <div
      class="sc-1fhg5t3-1 cJDFba"
    >
      <div
        class="sc-8tbhpn-3"
      >
        <div
          class="sc-1msmxxr-0 iWEzjW"
        />
        <table
          actions=""
          class="sc-1msmxxr-1 jwFsPF sc-8tbhpn-2 kCDmJI"
          columns="[object Object],[object Object],[object Object],[object Object],[object Object]"
          defaultactivepage="1"
          defaultsortorder="desc"
          itemsperpage="10"
          rowclickdisablekey=""
          tabindex="-1"
          totalitems="0"
        >
          <thead
            class="sc-uxzsxp-0 haNltq"
          >
            <tr
              class="sc-1jr91bg-0 hJzMcO"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            >
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Date
                  </span>
                  <svg
                    class="sc-1pf3x8a-0 bMQNVa"
                    fill="none"
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11.229 6.935a1 1 0 0 1 1.543 0l2.208 2.679a1 1 0 0 1-.77 1.636H9.79a1 1 0 0 1-.77-1.636l2.209-2.679zM12.771 17.065a1 1 0 0 1-1.543 0L9.02 14.386a1 1 0 0 1 .772-1.636h4.418a1 1 0 0 1 .771 1.636l-2.209 2.679z"
                      fill="#435465"
                    />
                  </svg>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Requested From
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Feedback About
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Status
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
              <th
                class="sc-hjmm4m-0 iOyNiA"
                tablesize="M"
              >
                <button
                  class="sc-hjmm4m-2 kpjakb"
                  tablesize="M"
                >
                  <span
                    class="sc-1ntgl3l-0 ghpJwn"
                    textalign="initial"
                    textcolor="inherit"
                  >
                    Action
                  </span>
                </button>
                <span
                  class="sc-hjmm4m-1 cuEDpa"
                />
              </th>
            </tr>
          </thead>
          <tbody
            class="sc-c7cniy-0 kShdfA"
          >
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="0"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="1"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="2"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="3"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
            <tr
              class="sc-1yxujaa-0 bgyIwA"
              gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
              style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
              tabindex="4"
            >
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
              <td
                class="sc-1e51eaa-0 hFOqbY"
                tablesize="M"
              >
                <div
                  class="sc-15vbyef-0 bHPECM"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </main>
</div>
`;
